[BB-724] Add setting to enable/disable Notes API and allow configuring notes as non-default site

xitij2000 opened the pull request:
Configuration Pull Request
---
This PR includes the following changes:

- It adds the ``EDX_NOTES_API_ENABLED`` setting to enable/disable the notes API app. 
- It adds the ``EDX_NOTES_API_HOSTNAME`` to support routing to the notes API by host name.
- It allows setting up the notes-api as a non-default site. 

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR if it is code shared across multiple services and you don't own all of the services.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

edx-webhook commented:
Thanks for the pull request, @xitij2000! I've created [OSPR-2934](https://openedx.atlassian.net/browse/OSPR-2934) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

natabene commented:
@xitij2000 Thank you for your contribution. @edx/arch-review Would you like to take a look before devops do?

natabene commented:
@ormsbee Thanks, will do!

xitij2000 commented:
@natabene Would love it get this looked at by devops.

ormsbee commented:
@edx/devops: Hi folks! Can someone please review this Notes API config PR?

natabene commented:
@edx/devops Can you give this a look?

xitij2000 commented:
@natabene Any updates :-)

natabene commented:
@edx/devops Could you give this a look when you have a chance? CC @bderusha

feanil commented on playbooks/roles/nginx/defaults/main.yml:
```diff
@@ -120,6 +120,8 @@ NGINX_EDXAPP_PROXY_INTERCEPT_ERRORS: false
 
 CMS_HOSTNAME: '~^((stage|prod)-)?studio.*'
 
+EDX_NOTES_API_HOSTNAME: '~^((stage|prod)-)?notes.*'
```
The nginx role is a bit weird in that most of the app profile related settings don't live in this role but in the relevant app role.  Not sure what the CMS_HOSTNAME is doing here but can you move `EDX_NOTES_API_HOSTNAME` default so that it lives in the `edx_notes_api` ansible role?

feanil commented on playbooks/edx_sandbox.yml:
```diff
@@ -63,7 +64,12 @@
+      when: EDX_NOTES_API_ENABLED
+    - role: nginx
+      nginx_sites:
+      - edx_notes_api
```
Can you add a comment here saying that we need to run nginx here again because the notes nginx profile is installed via the nginx role instead of by the app role like the other more modern services?

xitij2000 commented:
@feanil Thanks for the review! I will update the PR with the adjustments you recommended.

edx-webhook commented:
Thanks for the pull request, @xitij2000! I've created [OSPR-3482](https://openedx.atlassian.net/browse/OSPR-3482) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

natabene commented:
@feanil This is ready for your review.

feanil commented:
I'll get a second review on this from the devops team and then we can hopefully merge this.

xitij2000 commented:
Thanks!

adzuci commented:
Follow up comment, I didn't modify the https://tools-edx-jenkins.edx.org/job/sandboxes/job/CreateSandbox job to be able to include tools, but I did start to create a PR to add the variables, if possible we would like to make this available for developers to add to their future sandboxes.
https://github.com/edx/jenkins-job-dsl/pull/713

xitij2000 commented:
@adzuci Thanks! I have updated the PR based on your suggestions.  
I went with `EDX_NOTES_API_ENABLED` since I got an impression with the latest entry `JOURNALS_ENABLED` that this was the new format to go for :-)
I've also updated this in the changelog.

edx-webhook commented:
@xitij2000 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/configuration&entry.867055334=https://github.com/edx/configuration/pull/4913&entry.1484655318=https://github.com/xitij2000&entry.752974735=2018-12-27+16:07&entry.1917517419=2019-06-12+14:53&entry.2133058324=Yes) so we can improve your experience in the future.
